buffer.go

```diff
@@ -91,6 +92,11 @@ func (r *recordBuffer) Add(m *router.Message) error {
 		return err
 	}
 
+	// We default to a uuid if the template didn't match.
+	if pKey == "" {
+		pKey = uuid.New()
```

Comment:
Isn't the logging only in debug mode though?  Why not check this against `r.pKeyTmpl` directly?  Not sure what to do if the template fails to compile - my guess is crashing would be preferable to logging somewhere randomly?  maybe not.